CHANGES

```diff
@@ -12,6 +12,9 @@ Deprecated
 
 Features added
 --------------
+ 
+* #9800: extlinks: Emit warning if a hardcoded link is replaceable
+  by an extlink, suggesting a replacement.
```

Comment:
I think this improvements are enough stable (not a breaking change). So it would be better to release as 4.x. How about rebase this into the 4.x branch?